resources/main-template.cc

```diff
@@ -50,9 +51,54 @@ void TearDownOncePerProcess();
 }
 #endif
 namespace boxednode {
+namespace {
+struct TimingEntry {
+  const char* const label;
+  uint64_t const time;
+  TimingEntry* next = nullptr;
+  ~TimingEntry() {
+    delete next;
+  }
+};
+TimingEntry start_time_entry { "Process initialization", uv_hrtime() };
+std::atomic<TimingEntry*> current_time_entry { &start_time_entry };
+
+void MarkTime(const char* label) {
+  TimingEntry* new_entry = new TimingEntry { label, uv_hrtime() };
+  do {
+    new_entry->next = current_time_entry.load();
+  } while(!current_time_entry.compare_exchange_strong(new_entry->next, new_entry));
+}
+} // anonymous namespace
+
 Local<String> GetBoxednodeMainScriptSource(Isolate* isolate);
 Local<Uint8Array> GetBoxednodeCodeCacheBuffer(Isolate* isolate);
 std::vector<char> GetBoxednodeSnapshotBlobVector();
+
+void GetTimingData(const FunctionCallbackInfo<Value>& info) {
+  Isolate* isolate = info.GetIsolate();
+  TimingEntry* head = current_time_entry.load();
+  std::vector<Local<Value>> entries;
+  while (head != nullptr) {
+    Local<Value> elements[] = {
+      String::NewFromUtf8(isolate, head->label).ToLocalChecked(),
+      BigInt::NewFromUnsigned(isolate, head->time)
+    };
+    entries.push_back(Array::New(isolate, elements, sizeof(elements)/sizeof(elements[0])));
+    head = head->next;
+  }
+  Local<Array> retval = Array::New(isolate, entries.data(), entries.size());
+  info.GetReturnValue().Set(retval);
+}
+
+void boxednode_linked_bindings_register(
+    Local<Object> exports,
+    Local<Value> module,
+    Local<Context> context,
+    void* priv) {
+  NODE_SET_METHOD(exports, "getTimingData", GetTimingData);
```

Comment:
nit: Ignore me, I think getStartupTimingData would be more accurate? getTimingData is maybe too generic?

Reply:
For the current usage, yes, this would be right. But then the JS wrapper should probably still be named `getTimingData()` because it could be used for other things too, just with the current process start time as a baseline, so I think it's fine to keep these aligned?